### PR TITLE
CDAP-14827 add namespace generation to entity keys

### PR DESCRIPTION
**Pull request by albertshau**
Added namespace generation to entity keys to make it behave like
all workspaces and connections are deleted when a CDAP namespace
is deleted then re-created.

### CODE REVIEW

File: wrangler-proto/src/main/java/co/cask/wrangler/proto/connection/Connection.java [hunk 1]
```diff
@@ -54,9 +54,8 @@ public String getNamespace() {
   /**
    * @return id of the connection.
    */
-  public NamespacedId getId() {
```
Comment (CuriousVini) [inline wrangler-proto/src/main/java/co/cask/wrangler/proto/connection/Connection.java:57]: why this change is needed?

File: wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java [hunk 2]
```diff
@@ -123,6 +125,36 @@ protected Connection getValidatedConnection(ConnectionStore store, NamespacedId
...
+   * to create the response.
+   * If a {@link JsonSyntaxException} is thrown, a 400 response will be sent.
+   * If anything else if thrown, a 500 response will be sent.
+   * If nothing is thrown, the result of the callable will be sent as json.
+   *
+   * @param request the http request
+   * @param responder the http responder
+   * @param callable the endpoint logic to run
+   */
+  protected <T> void respond(HttpServiceRequest request, HttpServiceResponder responder, Callable<T> callable) {
```
Comment (CuriousVini) [inline wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java:143]: why do we need 2 methods one with namespace and one without?

File: wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java [hunk 3]
```diff
@@ -136,15 +168,19 @@ protected Connection getValidatedConnection(ConnectionStore store, NamespacedId
...
+                             NamespacedResponder<T> callable) {
     // system namespace does not officially exist, so don't check existence for system namespace.
-    if (namespace != null && !Contexts.SYSTEM.equals(namespace)) {
+    NamespaceSummary namespaceSummary;
+    if (Contexts.SYSTEM.equals(namespace)) {
+      namespaceSummary = new NamespaceSummary(Contexts.SYSTEM, "", 0L);
+    } else {
       try {
-        if (!getContext().getAdmin().namespaceExists(namespace)) {
+        namespaceSummary = getContext().getAdmin().getNamespaceSummary(namespace);
```
Comment (CuriousVini) [inline wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java:182]: is this call guaranteed to return latest generation of a namespace? Or only one generation is stored for a namespace at a given time?

**Review by CuriousVini — APPROVED** [review]
lgtm 👍  for code changes, just a few questions.

File: wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java (hunk 2 above)
Comment (albertshau) [reply wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java:143]: almost all endpoints are namespaced, but there are a couple that are not. They ones that are not use the method without a namespace, which avoids the unneeded lookup.

File: wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java (hunk 3 above)
Comment (albertshau) [reply wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java:182]: yep, there is only one at a given time.

File: wrangler-proto/src/main/java/co/cask/wrangler/proto/connection/Connection.java (hunk 1 above)
Comment (albertshau) [reply wrangler-proto/src/main/java/co/cask/wrangler/proto/connection/Connection.java:57]: This was only used for the "admin controlled" connections, which are specified in the app config (ConnectionTypeConfig). Those are connections that should be available in every namespace, so they only have an id and not a namespace.
